src/components/microlearning/data/hooks.js

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from 'react';
+import { logError } from '@edx/frontend-platform/logging';
+import { fetchVideoData } from './service';
+
+export function useVideoData(edxVideoID) {
+  const [videoData, setVideoData] = useState({});
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await fetchVideoData(edxVideoID);
```

Comment:
Aren't we supposed to use reactQuery for making API calls? I am asking this because we have shifted our learner portal API call structure to reactQuery.